presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArraySqlFunctions.java

```diff
@@ -89,4 +90,30 @@ public static String arrayHasDuplicatesVarchar()
     {
         return "RETURN cardinality(array_duplicates(input)) > 0";
     }
+
+    @SqlInvokedScalarFunction(value = "array_max_by", deterministic = true, calledOnNullInput = true)
+    @Description("Get the maximum value of array, by using a specific transformation function")
+    @TypeParameter("T")
+    @TypeParameter("U")
+    @SqlParameters({@SqlParameter(name = "input", type = "array(T)"), @SqlParameter(name = "f", type = "function(T, U)")})
+    @SqlType("T")
+    public static String arrayMaxBy()
+    {
+        return "RETURN input[" +
+                "array_max(zip_with(transform(input, f), sequence(1, cardinality(input)), (x, y)->IF(x IS NULL, NULL, (x, y))))[2]" +
+                "]";
+    }
+
+    @SqlInvokedScalarFunction(value = "array_min_by", deterministic = true, calledOnNullInput = true)
+    @Description("Get the minimum value of array, by using a specific transformation function")
+    @TypeParameter("T")
+    @TypeParameter("U")
+    @SqlParameters({@SqlParameter(name = "input", type = "array(T)"), @SqlParameter(name = "f", type = "function(T, U)")})
+    @SqlType("T")
+    public static String arrayMinBy()
+    {
+        return "RETURN input[" +
+                "array_min(zip_with(transform(input, f), sequence(1, cardinality(input)), (x, y)->IF(x IS NULL, NULL, (x, y))))[2]" +
+                "]";
```

Comment:
See my latest comment edit :) you don't need to index it even:

ARRAY_MIN(ZIP_WITH(TRANSFORM(input, f), input, (x, y)->x IS NOT NULL))[2]

Reply:
Oops:

ARRAY_MIN(ZIP_WITH(TRANSFORM(input, f), input, (x, y)->IF(x IS NULL, NULL, (x, y)))[2]

Reply:
There's still the issue with my previous reply ((link)[https://github.com/prestodb/presto/pull/18555#discussion_r1014524774]), if the input array isn't itself orderable (e.g. `array_min_by(map(...), ...)` this approach doesn't work

```
com.facebook.presto.sql.analyzer.SemanticException: Unexpected parameters (array(row(integer,map(varchar(3),integer)))) for function array_min. Expected: array_min(array(T)) T:orderable
```

Reply:
Ok Let's do the cardinality version.